torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -441,7 +441,9 @@ def _pre_forward_unshard(
     if not handle._prefetched:
         _unshard(state, handle, state._unshard_stream, state._pre_unshard_stream)
     handle._needs_pre_forward_unshard = False
-    state._device_handle.current_stream().wait_stream(state._unshard_stream)
+    # Don't wait during trace
```

Comment:
How is the trace valid if it contains stream information but not stream synchronization?
I guess we don't care for now since it's ignored by the backend but most likely will need fixing.